libvmi/core.c

```diff
@@ -367,9 +369,11 @@ set_id_and_name(
     if (VMI_SUCCESS == driver_get_name_from_id(vmi, id, &tmp_name)) {
         dbprint(VMI_DEBUG_CORE, "--got name from id (%"PRIu64" --> %s)\n", id, tmp_name);
         vmi->image_type = strndup(tmp_name, 100);
```

Comment:
The issue here that I mentioned had to do with the passing of tmp_name into strndup (a null-intolerant function) without checking after it was set in driver_get_name_from_id